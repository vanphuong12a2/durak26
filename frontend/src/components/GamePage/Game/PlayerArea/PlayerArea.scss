.player-area {
  align-self: center;

  .card-set {
    max-width: 80%;
    margin: auto;
    justify-content: center;
  }

  &.left {
    .player-info {
      text-align: right;
    }
    .card-set {
      padding-bottom: 50%;
      transform: rotate(90deg);
    }
  }

  &.right .card-set {
    padding-top: 50%;
    transform: rotate(90deg);
  }

  &.bottom {
    padding-right: 25%;
    padding-left: 25%;
  }
}
