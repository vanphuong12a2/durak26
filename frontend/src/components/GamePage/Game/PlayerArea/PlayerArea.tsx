import React from 'react';
import Player from '../../../../models/Player';
import {FaceDownCardSet} from '../CardSet/CardSet';
import './PlayerArea.scss';

interface Props {
  position: number
  player?: Player
}

const POSITIONS = ['bottom', 'top', 'left', 'right'];
const PlayerArea = (props: Props) => {

  const areaWithPlayer = (player: Player) => {
    return (
      <React.Fragment>
        <div className='player-info'>{player.name}</div>
        <FaceDownCardSet numberOfCards={player.numberOfCards}/>
      </React.Fragment>
    )
  };

  const areaWithoutPlayer = <React.Fragment/>;

  return (
    <div className={`player-area ${POSITIONS[props.position]}`}>
      {props.player ? areaWithPlayer(props.player) : areaWithoutPlayer}
    </div>
  );
};

export default PlayerArea;

