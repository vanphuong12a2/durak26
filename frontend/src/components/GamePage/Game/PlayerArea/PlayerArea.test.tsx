import React from 'react';
import {create} from 'react-test-renderer';
import PlayerArea from './PlayerArea';

describe('<PlayerArea />', () => {

  describe('Snapshots', () => {
    it('renders correctly', () => {
      const component = create(<PlayerArea position='top'/>);
      expect(component.toJSON()).toMatchSnapshot();
    });
  });

});
