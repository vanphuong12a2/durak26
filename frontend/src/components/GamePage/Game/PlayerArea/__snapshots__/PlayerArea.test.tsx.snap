// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PlayerArea /> Snapshots renders empty player area correctly 1`] = `
<div
  className="player-area right"
>
  <div
    className="player-info"
  >
    someName
    -
    d
  </div>
  <div
    className="card-set"
  >
    <img
      className="card"
      src="2D.svg"
    />
  </div>
</div>
`;

exports[`<PlayerArea /> Snapshots renders player area with cards correctly 1`] = `
<div
  className="player-area bottom"
>
  <div
    className="player-info"
  >
    someName
    -
    d
  </div>
  <div
    className="card-set"
  >
    <img
      className="card"
      src="2D.svg"
    />
  </div>
</div>
`;
